### PR TITLE
Add Error handling on duplicate dashboard names

### PR DESCRIPTION
**Pull request by rupal-bq**
### Description
- added check if dashboard with same name already exists

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/components/custom_panels/redux/panel_slice.ts [hunk 1]
```diff
@@ -141,6 +141,13 @@ export const uuidRx = /^[0-9a-fA-F]{8}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a
 
 export const isUuid = (id) => !!id.match(uuidRx);
 
+export const doesNameExist = (newCustomPanelName: string) => async() => {
+  const panels = await fetchCustomPanels();
+  if((panels.some((i: { title: string; }) => i.title === newCustomPanelName))){
```
Comment (joshuali925) [inline public/components/custom_panels/redux/panel_slice.ts:146]: nit. didn't run lint? missing space after `if`. Also probably `reutrn panels.some((i: { title: string; }) => i.title === newCustomPanelName)`?

and what happens if fetch call failed?

File: public/components/custom_panels/custom_panel_view_so.tsx [hunk 2]
```diff
@@ -208,7 +209,10 @@ export const CustomPanelViewSO = (props: CustomPanelViewProps) => {
   };
 
   const onRename = async (newCustomPanelName: string) => {
-    if (!isNameValid(newCustomPanelName)) {
+    const nameFlag = await doesNameExist(newCustomPanelName);
+    if (await nameFlag()) {
```
Comment (joshuali925) [inline public/components/custom_panels/custom_panel_view_so.tsx:213]: why await twice?

File: public/components/custom_panels/custom_panel_table.tsx [hunk 3]
```diff
@@ -120,7 +121,10 @@ export const CustomPanelTable = ({
   };
 
   const onCreate = async (newCustomPanelName: string) => {
-    if (!isNameValid(newCustomPanelName)) {
+    const nameFlag = await doesNameExist(newCustomPanelName);
+    if (await nameFlag()) {
```
Comment (pjfitzgibbons) [inline public/components/custom_panels/custom_panel_table.tsx:125]: The double-await seems weird?

Also nameFlag: might tell the story better if was "duplicateFound" or sth

File: public/components/custom_panels/custom_panel_table.tsx [hunk 4]
```diff
@@ -130,7 +134,10 @@ export const CustomPanelTable = ({
   };
 
   const onRename = async (newCustomPanelName: string) => {
-    if (!isNameValid(newCustomPanelName)) {
+    const nameFlag = await doesNameExist(newCustomPanelName);
+    if (await nameFlag()) {
```
Comment (pjfitzgibbons) [inline public/components/custom_panels/custom_panel_table.tsx:138]: Also double-wait

File: public/components/custom_panels/custom_panel_view_so.tsx [hunk 5]
```diff
@@ -208,7 +209,10 @@ export const CustomPanelViewSO = (props: CustomPanelViewProps) => {
   };
 
   const onRename = async (newCustomPanelName: string) => {
```
Comment (pjfitzgibbons) [inline public/components/custom_panels/custom_panel_view_so.tsx:211]: Why is `onRename` declared twice in different places?

File: .cypress/integration/3_panels.spec.ts [hunk 6]
```diff
@@ -134,6 +134,19 @@ describe('Testing panels table', () => {
       expectUuid(renamed);
     });
 
+    it.only('Renames panel with existing name', () => {
```
Comment (joshuali925) [inline .cypress/integration/3_panels.spec.ts:137]: remove `.only`?

File: public/components/custom_panels/redux/panel_slice.ts [hunk 7]
```diff
@@ -141,12 +141,10 @@ export const uuidRx = /^[0-9a-fA-F]{8}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a
...
 export const isUuid = (id) => !!id.match(uuidRx);
 
-export const doesNameExist = (newCustomPanelName: string) => async() => {
+export const doesNameExist = async (newCustomPanelName: string) => {
   const panels = await fetchCustomPanels();
-  if((panels.some((i: { title: string; }) => i.title === newCustomPanelName))){
-    return true;
-  }
-  return false;
+  console.log(panels);
```
Comment (joshuali925) [inline public/components/custom_panels/redux/panel_slice.ts:146]: remove debug message?

File: public/components/custom_panels/custom_panel_view_so.tsx (hunk 2 above)
Comment (rupal-bq) [reply public/components/custom_panels/custom_panel_view_so.tsx]: removed

File: public/components/custom_panels/custom_panel_table.tsx (hunk 3 above)
Comment (rupal-bq) [reply public/components/custom_panels/custom_panel_table.tsx]: removed

File: public/components/custom_panels/custom_panel_view_so.tsx (hunk 5 above)
Comment (rupal-bq) [reply public/components/custom_panels/custom_panel_view_so.tsx:211]: One is used in list view and another is details page

File: public/components/custom_panels/custom_panel_table.tsx (hunk 4 above)
Comment (rupal-bq) [reply public/components/custom_panels/custom_panel_table.tsx]: removed

File: .cypress/integration/3_panels.spec.ts (hunk 6 above)
Comment (rupal-bq) [reply .cypress/integration/3_panels.spec.ts]: removed

File: public/components/custom_panels/redux/panel_slice.ts (hunk 7 above)
Comment (rupal-bq) [reply public/components/custom_panels/redux/panel_slice.ts]: done

**Comment by rupal-bq** [comment]
https://user-images.githubusercontent.com/56703525/236319427-94e23e96-b43f-48a8-a682-ebbca579e07e.mov

**Review by mengweieric — APPROVED** [review]

**Review by ps48 — APPROVED** [review]

**Review by joshuali925 — APPROVED** [review]

**Pull request merged**